src/com/dojo/foundations/parking/TheGarage.java

```diff
@@ -89,81 +49,71 @@ public static void main(String[] args) throws Exception {
                     String truckPlate = scanner.readLine();
                     Matcher matcherTruck = PATTERN_TRUCK_PLATE.matcher(truckPlate);
-                    if (!matcherTruck.find()) {
-                        System.out.println("The plate is invalid");
-                        return;
-                    }
+                    evaluatePlate(matcherTruck);
                     System.out.println("Write the truck ton");
                     double truckTon = Double.parseDouble(scanner.readLine());
 
                     Optional<Cell> availableCellTruck = parkingLot.getParkingLot().stream()
-                            .filter(cell -> CellTypeEnum.TRUCK.equals(cell.getType()) && cell.isFree)
+                            .filter(cell -> CellTypeEnum.TRUCK.equals(cell.getType()) && cell.isFree())
                             .findAny();
 
                     if (availableCellTruck.isPresent()) {
                         availableCellTruck.get().setFree(false);
                         availableCellTruck.get().setEnterDateTime(Optional.of(LocalDateTime.now()));
                         parkingLotTruckMap.put(availableCellTruck.get(), new Truck(truckPlate, truckTon));
                     }
-                    parkingLotTruckMap.forEach((cell, truck) -> System.out.println("Cell: " + cell.code + " Plate: " + truck.plate));
+                    parkingLotTruckMap.forEach((cell, truck) -> System.out.println("Cell: " + cell.getCode() + " Plate: " + truck.getPlate()));
                     break;
                 case 4:
                     System.out.println("Write the motorcycle plate:");
                     String motorcycleLeavingPlate = scanner.readLine();
                     Matcher matcherLeavingMotorcycle = PATTERN_MOTORCYCLE_PLATE.matcher(motorcycleLeavingPlate);
-                    if (!matcherLeavingMotorcycle.find()) {
-                        System.out.println("The plate is invalid");
-                        return;
-                    }
+                    evaluatePlate(matcherLeavingMotorcycle);
                     Optional<Map.Entry<Cell, Motorcycle>> leavingMotorcycleMap = parkingLotMotorcycleMap
                             .entrySet().stream()
-                            .filter(entry -> motorcycleLeavingPlate.equals(entry.getValue().plate))
+                            .filter(entry -> motorcycleLeavingPlate.equals(entry.getValue().getPlate()))
                             .findFirst();
 
                     if (leavingMotorcycleMap.isPresent()) {
-                        double chargedAmountMotorcycle = calculateBillMotorcycle(leavingMotorcycleMap.get().getKey(), leavingMotorcycleMap.get().getValue());
+                        Cell leavingCell = leavingMotorcycleMap.get().getKey();
+                        Motorcycle leavingMotorcycle = leavingMotorcycleMap.get().getValue();
+                        double chargedAmountMotorcycle = leavingMotorcycle.calculateBilling(leavingCell.obtainHoursToCharge());
```

Comment:
I like what you did here.  Calculating the billing using the cell rater that the Cell/Vehicle map